catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/CsexraAttribute.java

```diff
@@ -0,0 +1,140 @@
+            new IsrAttributes()),
+    @SuppressWarnings("RedundantTypeArguments")
+    SNOW_DEPTH_MAX(Isr.SNOW_DEPTH_MAX_CENTIMETERS,
+            NitfConstants.SNOW_DEPTH_CAT,
+            getSnowDepthAccessorFunction(Pair<Double, Double>::getRight),
+            new IsrAttributes()),
+    PREDICTED_NIIRS(Isr.NATIONAL_IMAGERY_INTERPRETABILITY_RATING_SCALE,
+            NitfConstants.PREDICTED_NIIRS, tre -> {
+        return Optional.ofNullable(TreUtility.getTreValue(tre, NitfConstants.PREDICTED_NIIRS))
+            .filter(String.class::isInstance)
+            .map(String.class::cast)
+            .map(CsexraAttribute::parseNiirs)
+            .orElse(null);
+    }, new IsrAttributes()),
+    SNOW_COVER(Isr.SNOW_COVER, NitfConstants.GRD_COVER, tre -> {
+        return TreUtility.findIntValue(tre, NitfConstants.GRD_COVER)
+                .map(value -> {
+                    switch (value) {
+                    case 0:
+                        return Boolean.FALSE;
+                    case 1:
+                        return Boolean.TRUE;
+                    default:
+                        return null;
+                    }
+                })
+                .orElse(null);
+    }, new IsrAttributes());
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CsexraAttribute.class);
+
+    private String shortName;
+
+    private String longName;
+
+    private Function<Tre, Serializable> accessorFunction;
+
+    private AttributeDescriptor attributeDescriptor;
+
+    CsexraAttribute(String longName, String shortName, Function<Tre, Serializable> accessorFunction,
+            MetacardType metacardType) {
+        this.longName = longName;
+        this.shortName = shortName;
+        this.accessorFunction = accessorFunction;
+        // retrieving metacard attribute descriptor for this attribute to prevent later lookups
+        this.attributeDescriptor = metacardType.getAttributeDescriptor(longName);
+    }
+
+    private static Integer parseNiirs(String niirs) {
+        return NitfConstants.NIIRS_FORMAT.matcher(niirs)
+                .matches() ?
+                (int) Math.round(Double.valueOf(niirs)) :
+                null;
+    }
+
+    private static double inchesToCentimeters(double inches) {
+        return inches * 2.54;
+    }
+
+    private static Function<Tre, Serializable> getSnowDepthAccessorFunction(
+            Function<Pair, Double> pairFunction) {
+        return tre -> TreUtility.findIntValue(tre, NitfConstants.SNOW_DEPTH_CAT)
+                .map(CsexraAttribute::convertSnowDepthCat)
+                .map(doubleDoublePair -> doubleDoublePair.map(pairFunction)
+                        .orElse(null))
+                .orElse(null);
+    }
+
+    private static Optional<Pair<Double, Double>> convertSnowDepthCat(int category) {
+        switch (category) {
+        case 0:
+            return Optional.of(new ImmutablePair<>(inchesToCentimeters(0), inchesToCentimeters(1)));
```

Comment:
The idea is that category 0 represents the range. [0,1). But an arguments could be made for the ranges to be:

0 --> [0,0.5)
1 --> [0.5,8.5)
2 --> [8.5,17.5)
3 --> [17.5,MAX)